metagraph/src/annotation/taxonomy/tax_classifier.cpp

```diff
@@ -0,0 +1,199 @@
+#include "tax_classifier.hpp"
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "annotation/representation/annotation_matrix/annotation_matrix.hpp"
+#include "common/unix_tools.hpp"
+
+#include "common/logger.hpp"
+
+namespace mtg {
+namespace annot {
+
+using mtg::common::logger;
+
+TaxonomyClsAnno::TaxonomyClsAnno(const graph::AnnotatedDBG &anno,
+                                 const double lca_coverage_rate,
+                                 const double kmers_discovery_rate,
+                                 const std::string &tax_tree_filepath,
+                                 const std::string &label_taxid_map_filepath) : _anno_matrix(&anno) {
+    _lca_coverage_rate = lca_coverage_rate;
+    _kmers_discovery_rate = kmers_discovery_rate;
```

Comment:
```suggestion
                                 const std::string &label_taxid_map_filepath)
          : _anno_matrix(&anno),
            _lca_coverage_rate(lca_coverage_rate),
            _kmers_discovery_rate(kmers_discovery_rate) {
```

Reply:
I tried to implement this, but `_lca_coverage_rate` and `_kmers_discovery_rate` are inherited base class fields and got an error. Should I create a constructor for `TaxonomyBase` to initialize these two `_lca_coverage_rate` and `_kmers_discovery_rate`?

Reply:
If they're stored in the base class, then it's best to have a constructor for the base, then call the base's constructor here before setting `_anno_matrix`